### ISSUE TITLE
Relayer hangs on trying to send to a channel that isn't associated with the set connection

### ISSUE THREAD

**Issue by colin-axner** [cosmos/relayer#313]
Config:
```
global:
  timeout: 10s
  light-cache-size: 20
chains:
- key: str
  chain-id: stargate-4
  rpc-addr: http://35.223.51.79:26657
  account-prefix: cosmos
  gas-adjustment: 1.3
  gas-prices: 0.025umuon
  trusting-period: 336h
- key: slippy
  chain-id: slippy-0
  rpc-addr: http://localhost:26657
  account-prefix: cosmos
  gas-adjustment: 1.3
  gas-prices: 0.025slippers
  trusting-period: 167h
paths:
  cosmicslippers:
    src:
      chain-id: slippy-0
      client-id: thestargate
      connection-id: cosmicslippers0
      channel-id: slipperchannel
      port-id: transfer
      order: unordered
      version: ics20-1
    dst:
      chain-id: stargate-4
      client-id: slipperland
      connection-id: slipperportal0
      channel-id: theslipperchannel
      port-id: transfer
      order: unordered
      version: ics20-1
    strategy:
      type: naive
```

```
- channel_id: slipperchannel
  connection_hops:
  - cosmicslippers10
  counterparty:
    channel_id: theslipperchannel
    port_id: transfer
  ordering: ORDER_UNORDERED
  port_id: transfer
  state: STATE_OPEN
  version: ics20-1
```

Note that the connection_hops does not match the connection.

This cmd hangs:
```
rly transact transfer slippy-0 stargate-4 1000slippers cosmos1vte0c8t2c3ksn6q3j9km775wnnp303sc64z3lg -d
```

**Comment by colin-axner** [comment]
This also hangs on freshly created connection/channels, just tested

**Comment by colin-axner** [comment]
fixed in #314
